hyperspy/models/model1d.py

```diff
@@ -710,15 +710,36 @@ def _model2plot(self, axes_manager, out_of_range2nans=True):
             ns[np.where(self.channel_switches)] = s
             s = ns
         return s
-
-    def plot(self, plot_components=False, **kwargs):
+    
+    def _residual2plot(self, axes_manager, out_of_range2nans=True):
+        old_axes_manager = None
+        if axes_manager is not self.axes_manager:
```

Comment:
Could you document what happens here?

Reply:
I added some comment for what _residual_for_plot does.

Reply:
I have added documentation for _residual_for_plot.